Lib/enum.py

```diff
@@ -101,12 +100,29 @@ def __setitem__(self, key, value):
                 raise TypeError('%r already defined as: %r' % (key, self[key]))
             if isinstance(value, auto):
                 if value.value == _auto_null:
-                    value.value = self._generate_next_value(key, 1, len(self._member_names), self._last_values[:])
+                    value.value = self._generate_next_value(key, 1, len(self.members), list(self.members.values()))
                 value = value.value
-            self._member_names.append(key)
-            self._last_values.append(value)
+            self.members[key] = value
         super().__setitem__(key, value)
 
+    @property
```

Comment:
the deprecation part could be addressed in a separate commit.

Reply:
I don't quite understand how it should look like. Should I change anything now?